File: web/src/plugin/limitedActivity/api/banner.js
import service from '@/utils/request'

/** 限时活动 Banner 管理 */
export const createBanner = (data) =>
  service({ url: '/limitedActivity/banner/createBanner', method: 'post', data })
export const deleteBanner = (params) =>
  service({ url: '/limitedActivity/banner/deleteBanner', method: 'delete', params })
export const deleteBannerByIds = (data) =>
  service({ url: '/limitedActivity/banner/deleteBannerByIds', method: 'delete', data })
export const updateBanner = (data) =>
  service({ url: '/limitedActivity/banner/updateBanner', method: 'put', data })
export const findBanner = (params) =>
  service({ url: '/limitedActivity/banner/findBanner', method: 'get', params })
export const getBannerList = (params) =>
  service({ url: '/limitedActivity/banner/getBannerList', method: 'get', params })

File: server/plugin/limitedActivity/router/banner.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/limitedActivity/api"
	"github.com/gin-gonic/gin"
)

var apiBanner = api.Api.Banner

type bannerRouter struct{}

func (r *bannerRouter) Init(public, private *gin.RouterGroup) {
	g := private.Group("limitedActivity").Group("banner")
	g.Use(middleware.OperationRecord()).POST("createBanner", apiBanner.Create)
	g.Use(middleware.OperationRecord()).DELETE("deleteBanner", apiBanner.Delete)
	g.Use(middleware.OperationRecord()).DELETE("deleteBannerByIds", apiBanner.DeleteByIds)
	g.Use(middleware.OperationRecord()).PUT("updateBanner", apiBanner.Update)
	g.GET("findBanner", apiBanner.Find)
	g.GET("getBannerList", apiBanner.GetList)
}
